### DISCUSSION TITLE
A standalone Resource Server for standalone Aspire Dashboard

### DISCUSSION THREAD

**Discussion by kiapanahi** [microsoft/aspire#6772] [Show and tell]
There's a discussion going on about seeing and managing resources when using the standalone Aspire dashboard (e.g., running the dashboard as a container using the stock docker image, etc.) #4440 

In my case, I am working with a team of `PHP`, `Python`, and `Golang` developers who are trying to create a DevEnv on their local development machines to as closely as possible mimic the actual private cloud environment. The Aspire dashboard helps a lot but lacks one key aspect since it's not able to manage external resources created using multiple Docker Compose files without a running `ResourceServer`.

I'm not gonna mention everyone in said discussion but a bunch of people are already interested or even started working on a standalone resource server to point the dashboard to.

I've gone ahead and created a standalone resource server implementing the `proto` contract that the Aspire dashboard uses to communicate with the resource server implemented in the `Aspire.Hosting` project.

The code is in the early stages and [this is the repository I created for it](https://github.com/kiapanahi/Aspire.ResourceServer.Standalone).

I think with this asset, we can further the usage penetration among those who for any reason such as the one mentioned above, can't or won't commit to implementing the whole Application Model in .NET.

I'd appreciate any/all help or ideas about this thing.

as @davidfowl said in the linked discussion:

> [... omitted] Lets see if we can get people interested in this! This is great for the ecosystem!

Before I forget, Here's a picture of the standalone dashboard fetching Docker Container resources from the standalone dashboard

![image](https://github.com/user-attachments/assets/551d1f0d-8b3f-4b58-87b9-15f403db42b8)

**Comment by davidfowl** [comment]
If you combined this with https://github.com/prom3theu5/aspirational-manifests, the docker compose based deployment model would be complete ☺️

**Reply by kiapanahi** [reply]
Good idea

I've been swamped these past couple of weeks, but I'll give it a look ASAP. It seems promising.

**Comment by tjwald** [comment]
Would love to see this as a k8s controller!
